mmengine/dist/utils.py

```diff
@@ -94,6 +94,15 @@ def _init_dist_mpi(backend, **kwargs) -> None:
             'nccl', 'gloo' and 'mpi'. Defaults to 'nccl'.
         **kwargs: keyword arguments are passed to ``init_process_group``.
     """
+    if backend == 'smddp':
+        try:
+            import smdistributed.dataparallel.torch.torch_smddp  # noqa: F401
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                'Please use an Amazon SageMaker DLC to access smdistributed: '
+                'https://github.com/aws/deep-learning-containers/blob/master'
+                '/available_images.md#sagemaker-framework-containers'
+                '-sm-support-only') from e
```

Comment:
Hi! I have no experience with Amazon SageMaker, but is `smddp` only compatible with `mpi` launcher? If not, I think it's better to do the same thing in other launcher (e.g. `_init_dist_pytorch`), or maybe we can simply place this `import` statement into `init_dist`?

Reply:
Hi, from my understanding when smdistributed is enabled it is always launched with mpirun.

Reply:
Sounds reasonable